### ISSUE TITLE
Running NMS over EMA outputs generate a bunch of more-or-less-identical detections

### ISSUE THREAD

**Issue by tserdar** [ultralytics/yolov5#6570]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

When I run ema.ema() in eval model during training and apply non_maximum_suppression() afterwards I get many preds that are more or less near each other, very small and not overlapping. This does not happen when I run model() in eval mode under the same constraints for the same images.

Any ideas why?

**Comment by ouxiand** [comment]
How to solve this problem? I have the same problem.

**Comment by glenn-jocher** [comment]
@rocketsfallonrocketfalls did you put the EMA into .eval() mode?

**Comment by tserdar** [comment]
@ouxiand yes I have solved the issue but it may only apply to my specific case. I needed to use the EMA object before backward propagation. But I guess the EMA object in the script was not getting initialized/updated properly from a pretrained model before the first epoch ends. What I did was basically to assign the model to the EMA object in the code before the for loop of the epochs, and the issue is fixed.
